hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/builder/TargetFilterQueryCreate.java

```diff
@@ -50,19 +57,32 @@ default TargetFilterQueryCreate autoAssignDistributionSet(final DistributionSet
     }
 
     /**
+     * Set ID of {@link DistributionSet} for auto assignment
+     * 
      * @param dsId
      *            for {@link TargetFilterQuery#getAutoAssignDistributionSet()}
      * @return updated builder instance
      */
     TargetFilterQueryCreate autoAssignDistributionSet(Long dsId);
 
     /**
+     * Set {@link ActionType} for auto assignment
+     * 
      * @param actionType
      *            for {@link TargetFilterQuery#getAutoAssignActionType()}
      * @return updated builder instance
      */
     TargetFilterQueryCreate autoAssignActionType(ActionType actionType);
 
+    /**
+     * Set weight of {@link Action} created during auto assignment
+     * 
+     * @param weight
+     *            weight of {@link Action} generated within auto assignment
+     * @return updated builder instance
+     */
+    TargetFilterQueryCreate autoAssignWeight(Integer weight);
```

Comment:
Yes.... most of the auto assign attributed methods are used just in test classes. That being said, since this is a builder, and we intend to avail that flexibility here, I would rather leave it unharmed :D